### PR TITLE
Convert ui.test_activationkey to VMBroker

### PR DESCRIPTION
**Pull request by mshriver**
Mostly using existing features, used VMBroker directly for multi-count
test coverage

Also adjusted the imports

Test results are consistent with master branch - there are test cases failing because of non-interactable elements in search.

### CODE REVIEW

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_activationkey.py

**Review by swadeley — APPROVED** [review]
ACK

File: tests/foreman/ui/test_activationkey.py [hunk 1]
```diff
@@ -111,19 +96,19 @@ def test_positive_end_to_end_register(session):
...
-        repos_collection.setup_virtual_machine(vm)
-        with session:
-            session.organization.select(org.name)
-            chost = session.contenthost.read(vm.hostname, widget_names='details')
-            assert chost['details']['registered_by'] == f'Activation Key {ak_name}'
-            ak_values = session.activationkey.read(ak_name, widget_names='content_hosts')
-            assert len(ak_values['content_hosts']['table']) == 1
-            assert ak_values['content_hosts']['table'][0]['Name'] == vm.hostname
+
+    repos_collection.setup_virtual_machine(rhel7_contenthost)
```
Comment (mshriver) [inline tests/foreman/ui/test_activationkey.py:104]: I'll be changing this again in #8665 as this method gets moved to `ContentHost` class - but for now, it's like this.

Comment (mshriver) [reply tests/foreman/ui/test_activationkey.py:104]: I was thinking of a different method - this one from `robottelo.products.RepositoryCollection` stays.

File: tests/foreman/ui/test_activationkey.py [hunk 2]
```diff
@@ -1109,36 +1097,38 @@ def test_positive_service_level_subscription_with_custom_product(session):
...
-            subscriptions = {
-                subs['Repository Name'] for subs in chost['subscriptions']['resources']['assigned']
-            }
-            assert product.name in subscriptions
+
+    rhel7_contenthost.install_katello_ca()
+    rhel7_contenthost.register_contenthost(org.label, activation_key=activation_key.name)
+    assert rhel7_contenthost.subscribed
+    result = rhel7_contenthost.run('subscription-manager list --consumed')
+    assert result.return_code == 0
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_activationkey.py:1123]: I believe this should check status, `return_code` doesn’t exist 
```suggestion
    assert result.status == 0
```

**Review by JacobCallahan — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

**Pull request merged**